app/views/documents/show/_history_tab.html.erb

```diff
@@ -11,6 +11,14 @@
       <%= t "documents.history.entry_types.#{entry.entry_type}" %>
     </h4>
 
+    <div class="app-timeline-entry__dateline">
+      <%= entry.created_at.strftime("%d %B %Y at %l:%M%P") %> by <%= entry.username_or_unknown %>
+    </div>
+
+    <% if entry.internal_note.present? %>
```

Comment:
I imagine by accessing these this way we have an [n+1 issue](https://semaphoreci.com/blog/2017/08/09/faster-rails-eliminating-n-plus-one-queries.html) - which I guess we probably already had with removal and retire, it'd be good if we could try resolve that.

Reply:
Yep sure, i'll look into it